src/endpoints/authentication/cose_auth.cpp

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "ccf/endpoints/authentication/cose_auth.h"
+
+#include "ccf/crypto/cose_verifier.h"
+#include "ccf/crypto/public_key.h"
+#include "ccf/crypto/verifier.h"
+#include "ccf/http_consts.h"
+#include "ccf/pal/locking.h"
+#include "ccf/rpc_context.h"
+#include "ccf/service/tables/members.h"
+#include "ccf/service/tables/users.h"
+#include "ds/lru.h"
+#include "http/http_sig.h"
+
+#include <qcbor/qcbor.h>
+#include <qcbor/qcbor_spiffy_decode.h>
+#include <t_cose/t_cose_sign1_verify.h>
+//#include <t_cose/t_cose_common.h>
+
+namespace ccf
+{
+  namespace cose
+  {
+    static constexpr int64_t COSE_HEADER_PARAM_ALG = 1;
+    static constexpr int64_t COSE_HEADER_PARAM_KID = 4;
+    // static constexpr const char* COSE_HEADER_PARAM_??? = "";
+    // static constexpr const char* COSE_HEADER_PARAM_??? = "";
```

Comment:
nit: remove?

Reply:
Fixed now, that's the headers we need to identify our proposals.